business_rules/engine.py

```diff
@@ -105,7 +105,5 @@ def fallback(*args, **kwargs):
             raise AssertionError("Action {0} is not defined in class {1}"\
                     .format(method_name, defined_actions.__class__.__name__))
         params = action.get('params') or {}
-        params["results"] = results
```

Comment:
Since params are taken from actions argument which is being passed by reference, the original rule dict was affected by this assignment and rule["actions"]["params"] also had results key.

Reply:
Does this break reporting? If so we should just copy the params and add the results